internal/sim/api.go

```diff
@@ -192,28 +225,52 @@ type Results struct {
 }
 
 // New returns a new Simulator that simulates workload W.
-func New[W any, P WorkloadPointer[W]](t *testing.T) *Simulator {
+func New[W any, P WorkloadPointer[W]](t *testing.T, opts Options) *Simulator {
 	// Note that because the Init method on a workload struct T often has a
 	// pointer receiver *T, it is the pointer *T (rather than T itself) that
 	// implements the Workload interface.
 	t.Helper()
 
+	// Parse config.
+	app := &protos.AppConfig{}
+	if opts.Config != "" {
+		var err error
+		app, err = runtime.ParseConfig("", opts.Config, codegen.ComponentConfigValidator)
+		if err != nil {
+			t.Fatalf("sim.New: parse config: %v", err)
+		}
+	}
+
 	// Validate the workload struct.
 	w := reflection.Type[P]()
 	if err := validateWorkload(w); err != nil {
 		t.Fatalf("sim.New: invalid workload type %v: %v", w, err)
 	}
 
-	// Call Init. Validate the registered fakes and generators.
+	// Gather the set of registered components.
+	regs := codegen.Registered()
+	regsByIntf := map[reflect.Type]*codegen.Registration{}
+	for _, reg := range regs {
+		regsByIntf[reg.Iface] = reg
+	}
+
+	// Call Init and validate the registered fakes and generators.
+	//
+	// A simulator calls a workload's Init method many times, once per
```

Comment:
I presume we create new workload instances and call Init once per instance?

Do you know how many workload instances we plan to create, and how much the repeated validation will contribute to the cost compared to actual simulation runs? It feels like the API is slightly broken if we have to do this type of caching re: registrations. If anything, I would err on the side of repeated work.